generators/generator-base.js

```diff
@@ -2267,6 +2267,153 @@ templates: ${JSON.stringify(existingTemplates, null, 2)}`;
         );
   }
 
+  /**
+   * write the given files using provided options.
+   *
+   * @param {object} options
+   * @param {object} options.files - files to write
+   * @param {object} [options.context] - context for templates
+   * @param {string|string[]} [options.rootTemplatesPath] - path(s) to look for templates.
+   *        Single absolute path or relative path(s) between the templates folder and template path.
+   * @return {string[]}
+   */
+  async writeFiles(options) {
```

Comment:
Can it be put in a dedicated files and called from there? 
for example:
* `utils/generation/processing/template-processor.js // for ejs`
* `utils/generation/processing/file-processor.js // for svg, png, or other static files` 
* `utils/generation/processing/templates-processor.js // calling the two above`

Ideally, if we could make the classes SOLID (including the S) we could get some new opportunity

Reply:
IMO this will make more complicated.
I will try to make writeFiles more linear.